tests/api-watch-diff/test.ml

```diff
@@ -2,3 +2,214 @@ let%expect_test "test_diff_interface" =
+                  ~level:0 ~scope:0 ~id:0));
+        type_variance = [];
+        type_separability = [];
+        type_is_newtype = false;
+        type_expansion_scope = 0;
+        type_loc = Location.none;
+        type_attributes = [];
+        type_immediate = Unknown;
+        type_unboxed_default = false;
+        type_uid = Uid.internal_not_actually_unique;
+      },
+      Trec_not,
+      Exported )
+
+(* Signature for type t:
+     > type unused_type = string *)
+
+let unused_type_sig =
+  Sig_type
+    ( Ident.create_persistent "unused_type",
+      {
+        type_params = [];
+        type_arity = 0;
+        type_kind = Type_abstract;
+        type_private = Public;
+        type_manifest =
+          Some
+            (Transient_expr.type_expr
+               (Transient_expr.create
+                  (Tconstr
+                     (Path.Pident (Ident.create_predef "string"), [], ref Mnil))
+                  ~level:0 ~scope:0 ~id:1));
+        type_variance = [];
+        type_separability = [];
+        type_is_newtype = false;
+        type_expansion_scope = 0;
+        type_loc = Location.none;
+        type_attributes = [];
+        type_immediate = Unknown;
+        type_unboxed_default = false;
+        type_uid = Uid.internal_not_actually_unique;
+      },
+      Trec_not,
+      Exported )
+
+(* Signature for value f:
+    > val f : t -> string *)
+
+let val_f_sig =
+  Sig_value
+    ( Ident.create_persistent "f",
+      {
+        val_type =
+          Transient_expr.type_expr
+            (Transient_expr.create
+               (Tarrow
+                  ( Nolabel,
+                    Transient_expr.type_expr
+                      (Transient_expr.create
+                         (Tconstr
+                            (Path.Pident (Ident.create_predef "t"), [], ref Mnil))
+                         ~level:0 ~scope:0 ~id:2),
+                    Transient_expr.type_expr
+                      (Transient_expr.create
+                         (Tconstr
+                            ( Path.Pident (Ident.create_predef "string"),
+                              [],
+                              ref Mnil ))
+                         ~level:0 ~scope:0 ~id:3),
+                    commu_ok ))
+               ~level:0 ~scope:0 ~id:4);
+        val_kind = Val_reg;
+        val_loc = Location.none;
+        val_attributes = [];
+        val_uid = Uid.internal_not_actually_unique;
+      },
+      Exported )
+
+(* Signature for value g:
+    > val g : t -> t *)
+
+let val_g_sig =
+  Sig_value
+    ( Ident.create_persistent "g",
+      {
+        val_type =
+          Transient_expr.type_expr
+            (Transient_expr.create
+               (Tarrow
+                  ( Nolabel,
+                    Transient_expr.type_expr
+                      (Transient_expr.create
+                         (Tconstr
+                            (Path.Pident (Ident.create_predef "t"), [], ref Mnil))
```

Comment:
I think you shouldn't be using `create_predef` here. `t` is not a predefined OCaml type.